cmd/cli/exec/exec.go

```diff
@@ -223,129 +220,27 @@ func PrepareJob(cmd *cobra.Command, cmdArgs []string, unknownArgs []string, opti
 		}
 	}
 
-	publisherSpec := options.SpecSettings.Publisher.Value()
-	if publisherSpec != nil {
-		job.Tasks[0].Publisher = &models.SpecConfig{
-			Type:   publisherSpec.Type.String(),
-			Params: publisherSpec.Params,
-		}
-	}
-
-	// Handle ResultPaths by using the legacy parser and converting.
-	if err := prepareJobOutputs(cmd.Context(), options, job); err != nil {
-		return nil, err
-	}
-
-	// Parse labels from flag, we expect key=value for the non-legacy models.Job
-	if err := prepareLabels(options, job); err != nil {
-		return nil, err
-	}
-
-	// Constraints for node selection
-	if err := prepareConstraints(options, job); err != nil {
-		return nil, err
+	job.Labels, err = options.JobSettings.Labels()
+	job.Task().Publisher = options.TaskSettings.Publisher.Value()
+	job.Task().ResultPaths = options.TaskSettings.ResultPaths
+	job.Task().Env = options.TaskSettings.EnvironmentVariables
+	job.Task().InputSources = options.TaskSettings.InputSources.Values()
+	if err != nil {
+		return nil, fmt.Errorf("parsing job labes: %w", err)
```

Comment:
Why you are not using `BuildJobFromFlags` similar to `docker run` and `wasm run`, and then override whatever you need to override?

Reply:
I decided against this as the base version of the job spec is derived from a template and it was simpler to produce the templated job first then modify it.